### PR TITLE
Register twice disable

### PR DESCRIPTION
**Pull request by Minglee01**
@incuna/frontend  Stop register code running more than once.

### CODE REVIEW

File: src/registration/scripts/directives.js [hunk 1]
```diff
@@ -20,41 +21,45 @@
                         });
 
                     scope.register = function () {
-                        optionsPromise
-                            .then(function () {
-                                scope.loading = true;
-                                scope.registered = false;
+                        if (scope.clicked == false) {
```
Comment (wytrych) [inline src/registration/scripts/directives.js]: ```if (!scope.clicked)``` instead ?

Comment (Minglee01) [inline src/registration/scripts/directives.js]: will do im just used to writing long hand

**Comment by perry** [comment]
This seems like functionality already provided by the `scope.loading` flag. You could block interaction with the form when `scope.loading === true`.

**Comment by Minglee01** [comment]
How does that work i mean scope.loading isnt being defined anywhere. so
would it work outside the options promise.

**Comment by perry** [comment]
You could disable the submit button on the form like so `<button type="submit" ng-disabled="loading">Submit</button>`.

`scope.loading` will evaluate to being falsey when initially undefined.

**Comment by Minglee01** [comment]
so i want ng-disabled="!loading"
when it is loading the condition should switch to true and then disable the
button

**Comment by Minglee01** [comment]
ignore that

**Comment by perry** [comment]
You want `ng-disabled="loading"` which adds the `disabled` attribute when true.

**Comment by Minglee01** [comment]
yeah i said ignore me. I read the line the template and it hit.

**Comment by perry** [comment]
Okay, no problem.

@Minglee01 I've squashed your commits in to a single commit, and added the code to all the other directives in the library.

**Comment by perry** [comment]
I've also pushed the dist files in to that commit.

**Comment by Minglee01** [comment]
Cheers Perry good to merge?

**Comment by maciejpi** [comment]
@perry come back!

**Comment by perry** [comment]
I'm happy!

**Comment by maciejpi** [comment]
I hope so :)

**Comment by perry** [comment]
That was directed at Ming :trollface:

**Comment by maciejpi** [comment]
hahahahaha

**Comment by maciejpi** [comment]
I thought you said you are happy with your current work

**Comment by pandalion** [comment]
@perry PERRRRRRRY :panda_face: :heart:  :pear: :bear:

**Pull request merged**